docs/modules/ROOT/pages/references/projectsyn/developer.adoc

```diff
@@ -0,0 +1,125 @@
+
+This page defines the various conventions and best practices for writing a https://syn.tools/commodore/index.html[Commodore component] for OpenShift 4.
+
+[NOTE]
+====
+See the https://syn.tools/commodore/writing-a-component.html[Writing a Commodore Component] guide for a more general introduction and technical details of Commodore components.
+====
+
+
+== Component Naming
+
+The name of a component which configures something OpenShift 4 specific must be prefixed with `openshift4-`.
+
+[NOTE]
+====
+When using the `commodore component new` command to bootstrap a new component, the `component-` prefix will be added automatically and doesn't need to be specified:
+
+[source,console]
+----
+commodore component new openshift4-registry --name "OpenShift 4 Registry" --owner appuio
+----
+====
+
+
+== https://docs.openshift.com/container-platform/latest/operators/understanding_olm/olm-understanding-olm.html[Operator Lifecycle Manager (OLM)]
+
+We try to avoid using the OLM to install operators on a cluster.
+This is mainly due to the added complexity the OLM introduces and us not using any of it's features. This might change in the future.
+
+Exceptions to this rule include cases where the Red Hat documentation explicitly requires the OLM.
+
+Any operators we **do** install with the OLM should be restricted to single namespaces and must never be installed using the `AllNamespaces` InstallMode.
+
+We don't provide any support for operators installed by customers via the OLM.
+
+
+== CustomResource (CR) Specs
+
+Components which configure one or multiple CRs should do so by exposing its `.spec` field as a parameter.
+This enables the configuration of all possible fields via the inventory and provides the maximum flexibility.
+Default values can either be set in the `defaults.yml` file of the component or in JSONNET code.
+It should always be possible to override default values in the inventory.
+
+If creating multiple CRs, a dict should be used where its keys map to the names used for the CRs.
+Iterating over the keys of this dict in JSONNET enables the enrichment of possible default values if not set.
+
+Due to the way http://reclass.pantsfullofunix.net/index.html[reclass] handels `null` overrides, the component must handle `null` values in dicts gracefully.
+
+.main.jsonnet
+[source,jsonnet]
+----
+local ingressControllers =
+  if params.ingressControllers != null
+  then std.objectFields(params.ingressControllers)
+  else [];
+
+{
+  [name]:
+    [kube._Object('operator.openshift.io/v1', 'IngressController', name) {
+      metadata+: {
+        namespace: params.namespace,
+      },
+      spec: {
+        defaultCertificate: {
+          name: acmeCertName,
+        },
+      } + params.ingressControllers[name],
+    }]
+  for name in ingressControllers
+} + {
+  [if std.length(ingressControllers) == 0 then '.gitkeep']: {},
+}
+----
+
+
+== Inventory Parameters
+
+A component can define various parameters in the inventory (`defaults.yml`).
```

Comment:
This is not OpenShift specific and duplicates what would have do be documented by Syn. Assuming the component got bootstrapped with `commodore new component`, this is already a given.

Reply:
Yes, my thoughts exactly. But I thought better have it documented here first rather than nowhere. We can always move it again. WDYT?

Reply:
In that case, I would more focus on using the `new component` command than explaining how the naming is supposed to be. Or at least say something along the line: use that and this is solved for you.